app/lib/core/view_models/forgot_password_view_model.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:app/core/services/auth_service.dart';
+import 'package:app/core/services/dialog_service.dart';
+import 'package:app/core/services/nav_service.dart';
+import 'package:app/core/services/service_locator.dart';
+import 'package:app/core/view_models/base_view_model.dart';
+import 'package:app/ui/common/view_state.dart';
+import 'package:app/ui/views/check_your_email_account_screen.dart';
+import 'package:app/ui/views/sign_in_screen.dart';
+import 'package:flutter/material.dart';
+
+class ForgotPasswordViewModel extends BaseViewModel {
+  final DialogService _dialogService = locator<DialogService>();
+  final AuthenticationService _authenticationService =
+      locator<AuthenticationService>();
+  final NavigationService _navigationService = locator<NavigationService>();
+
+  void navigateToSignInScreen() =>
```

Comment:
- [ ] If we're coming from the SignInScreen to this screen, we should just `navigateBackTo` and not nav and replace. 

Nav and replace is meant for parallel screens. This screen is more meant for "sitting on top"